views/role.go

```diff
@@ -87,18 +90,94 @@ func (v *Views) RoleFunc(c echo.Context) error {
 		return fmt.Errorf("failed to get user permissions for role: %w", err)
 	}
 
+	permissions, err := v.user.GetPermissionsNotInRole(c.Request().Context(), role1)
+	if err != nil {
+		return fmt.Errorf("failed to get permissions not in role for role: %w", err)
+	}
+
+	users, err := v.user.GetUsersNotInRole(c.Request().Context(), role1)
+	if err != nil {
+		return fmt.Errorf("failed to get users not in role for role: %w", err)
+	}
+
 	data := RoleTemplate{
-		Role:            roleTemplate,
-		UserPermissions: p1,
-		ActivePage:      "role",
+		Role:                 roleTemplate,
+		PermissionsNotInRole: permissions,
+		UsersNotInRole:       users,
+		TemplateHelper: TemplateHelper{
+			UserPermissions: p1,
+			ActivePage:      "role",
+		},
+	}
+
+	return v.template.RenderTemplate(c.Response(), data, templates.RoleTemplate, templates.RegularType)
+}
+
+func (v *Views) roleFunc(c echo.Context, roleID int) error {
```

Comment:
Can this function have a more descriptive name?

Reply:
I just removed it, turns out it also has other issues that refreshing a page won't work properly